### DISCUSSION TITLE
How to simplify the task of write to_json function for derived class ?

### DISCUSSION THREAD

**Discussion by gzliudan** [nlohmann/json#2223] [Q&A] [answered]
If I have many derived classes which inherits the same parent class, how to simplify tasks of write to_json code for parent class?  

```cpp
class Base {};
void to_json(json& j, const Base& obj);

class Derived1 : public Base {}
void to_json(json& j, const Derived1& obj);

class Derived2 : public Base {}
void to_json(json& j, const Derived2& obj);
```

**Comment by nlohmann** [comment, marked as answer]
If some of the classes share code, you can call `to_json` of the base class explicitly, e.g. by

```cpp
nlohmann::to_json(j, static_cast<Base&>(obj));
```

Further note there is an [`update`](https://nlohmann.github.io/json/doxygen/classnlohmann_1_1basic__json_a6c02637be3d8bd730a14082cc05f7a85.html#a6c02637be3d8bd730a14082cc05f7a85) function to merge/update objects.

A related issue: #2199

**Comment by gzliudan** [comment]
Thank you very much!
